code/modules/research/designs.dm

```diff
@@ -604,7 +604,45 @@
 	build_path = /obj/item/weapon/gun/energy/cutter/plasma
 	sort_string = "TAEAC"
 
-
+/datum/design/item/weapon/divet
+	name = "Winchester NK Divet pistol"
+	id = "divet"
+	req_tech = list(TECH_COMBAT = 4, TECH_MATERIAL = 5)
+	materials = list(MATERIAL_STEEL = 10000, MATERIAL_SILVER = 4000)
+	build_path = /obj/item/weapon/gun/projectile/divet
+	sort_string = "TACEA"
+
+/datum/design/item/weapon/divetslug
+	name = "divet magazine"
+	id = "divetslug"
+	req_tech = list(TECH_COMBAT = 4, TECH_MATERIAL = 4)
+	materials = list(MATERIAL_STEEL = 4000)
+	build_path = /obj/item/ammo_magazine/divet
+	sort_string = "TACEB"
+
+/datum/design/item/weapon/50cal
+	name = ".50 AE magazine"
+	id = "50cal"
+	req_tech = list(TECH_COMBAT = 6, TECH_MATERIAL = 6, TECH_ENGINEERING = 6)
+	materials = list(MATERIAL_STEEL = 4000, MATERIAL_GOLD = 2000, MATERIAL_DIAMOND = 1000, MATERIAL_PLASTEEL = 2000) // Plasteel can be changed to deuterium later when I get around to defining material for it. If you have an idea for a reagent that could be used to minimize production of these, please add it.
+	build_path = /obj/item/ammo_magazine/a50
+	sort_string = "TACEH"
+
+/datum/design/item/weapon/pulserifle
+	name = "SWS Motorized Pulse Rifle"
+	id = "pulserifle"
+	req_tech = list(TECH_COMBAT = 5, TECH_MATERIAL = 6, TECH_ENGINEERING = 6)
```

Comment:
Tech Engineering hasn't to do anything with this. Tech combat suffices for weapons research, you do not need unrelated engineering research.
4 and 5 for tech combat and material please.